src/rgw/driver/sfs/types.cc

```diff
@@ -300,25 +383,37 @@ void Bucket::_undelete_object(ObjectRef objref, const rgw_obj_key & key,
   }
 }
 
+std::optional<ObjectRef> Bucket::get_from_db(const std::string& name) {
+  sqlite::SQLiteObjects objs(store->db_conn);
+  auto obj = objs.get_object(info.bucket.bucket_id, name);
+  if (!obj) {
+    return nullptr;
+  }
+  sqlite::SQLiteVersionedObjects objs_versions(store->db_conn);
+  auto last_version = objs_versions.get_last_versioned_object(obj->uuid);
+  if (last_version.has_value()) {
+    auto new_obj = Object::create_for_query(
+        obj->name, obj->uuid,
+        (last_version->object_state == ObjectState::DELETED), last_version->id);
+    new_obj->update_meta({.size = obj->size,
+                          .etag = obj->etag,
+                          .mtime = obj->mtime,
+                          .set_mtime = obj->set_mtime,
+                          .delete_at = obj->delete_at});
+    new_obj->update_attrs(last_version->attrs);
+    new_obj->instance = last_version->version_id;
+    return std::shared_ptr<Object>(std::move(new_obj));
+  }
+  return nullptr;
```

Comment:
same as above.

Reply:
Fixed. The methods now return a pointer not std::optional